### PR TITLE
RUMM-2889 Support untracked modals

### PR DESCRIPTION
**Pull request by fuzzybinary**
### What and why?

This allows modal views to be "untracked" (not call startView / stopView) when using automatic view tracking. The view disappearing under the modal will still call `stopView`, and will restart when the modal is dismissed.  This is useful for  when a user wants to manually track a view controller that is presented modally, or when a framework (such as Flutter or React Native) wants to take over view reporting.

### How?

Pre iOS 13, users will need to set `isUntrackedModal` on the view they return from the predicate. This will push the view onto the `RUMViewsHandler` stack, but this view will not be started or stopped by the `RUMViewsHandler`.

On iOS 13+, views that are marked with `isModalInPresentation`, but return null from the `rumViewPredicate` are automatically considered to be untracked modals.

### Review checklist
- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)
- [ ] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [x] Run unit tests
- [x] Run integration tests
- [ ] Run smoke tests

### CODE REVIEW

**Review by maxep — COMMENTED** [review]
Looks great, thanks! I left 2 minor comments.
Do you think it would be valuable to have an Integration test scenario where we randomly pick a modal presentation style and we assert that the ignored modal stops the previous view?

**Comment by fuzzybinary** [comment]
Yes, agree that's probably a good integration test. I'll add it.

File: Sources/Datadog/RUM/Instrumentation/Views/RUMViewsHandler.swift [hunk 1]
```diff
@@ -103,11 +106,16 @@ internal final class RUMViewsHandler {
 
         // Stop the last appearing view of the stack
         if let current = stack.last {
-            stop(view: current)
+            if !current.isUntrackedModal {
```
[the remark this answers is not in the document]
Comment (fuzzybinary) [reply Sources/Datadog/RUM/Instrumentation/Views/RUMViewsHandler.swift:109]: Yeah, I can move them it into start / stop if you think that's better.

File: Datadog/Example/Scenarios/RUM/RUMScenarios.swift [hunk 2]
```diff
@@ -74,6 +74,38 @@ final class RUMModalViewsAutoInstrumentationScenario: TestScenario {
     private class Predicate: UIKitRUMViewsPredicate {
         func rumView(for viewController: UIViewController) -> RUMView? {
             if let viewName = viewController.accessibilityLabel {
+                print(viewName)
```
Comment (maciejburda) [inline Datadog/Example/Scenarios/RUM/RUMScenarios.swift:77]: Probably a leftover `print()` 🧹

File: Datadog/Example/Scenarios/RUM/RUMScenarios.swift [hunk 3]
```diff
@@ -74,6 +74,38 @@ final class RUMModalViewsAutoInstrumentationScenario: TestScenario {
...
+                return nil
+            }
+        }
+    }
+
+    func configureSDK(builder: Datadog.Configuration.Builder) {
+        _ = builder
+            .trackUIKitRUMViews(using: Predicate())
+            .enableLogging(false)
+            .enableTracing(false)
```
Comment (maciejburda) [inline Datadog/Example/Scenarios/RUM/RUMScenarios.swift:89]: Side note:

Would be great to improve the API in V2. IMO it should be `.loggingEnabled(true)` an so on. cc @maxep

File: Datadog/Example/Scenarios/RUM/RUMScenarios.swift [hunk 4]
```diff
@@ -74,6 +74,38 @@ final class RUMModalViewsAutoInstrumentationScenario: TestScenario {
...
             if let viewName = viewController.accessibilityLabel {
+                print(viewName)
+                return .init(name: viewName)
+            } else {
+                return nil
+            }
+        }
+    }
+
+    func configureSDK(builder: Datadog.Configuration.Builder) {
```
Comment (maciejburda) [inline Datadog/Example/Scenarios/RUM/RUMScenarios.swift:85]: I don't think it's used 🤔

File: Datadog/Example/Scenarios/RUM/RUMScenarios.swift [hunk 5]
```diff
@@ -74,6 +74,38 @@ final class RUMModalViewsAutoInstrumentationScenario: TestScenario {
...
+/// Tracks view controllers as RUM Views.
+final class RUMUntrackedModalViewsAutoInstrumentationScenario: TestScenario {
+    static var storyboardName: String = "RUMModalViewsAutoInstrumentationScenario"
+
+    private class Predicate: UIKitRUMViewsPredicate {
+        func rumView(for viewController: UIViewController) -> RUMView? {
+            if let viewName = viewController.accessibilityLabel {
+                if viewController.modalPresentationStyle == .fullScreen {
+                    if #available(iOS 13, *) {
+                        // Untracked on iOS 13 via isModalInPresentation
```
Comment (maciejburda) [inline Datadog/Example/Scenarios/RUM/RUMScenarios.swift:103]:
```suggestion
                        // Untracked on iOS 13+ via isModalInPresentation
```

**Review by maciejburda — APPROVED** [review]

File: Datadog/Example/Scenarios/RUM/RUMScenarios.swift (hunk 4 above)
Comment (fuzzybinary) [reply Datadog/Example/Scenarios/RUM/RUMScenarios.swift:85]: It is. It's called after the scenario boots. For some reason you can't set a breakpoint in it though.

**Review by maxep — APPROVED** [review]
Nice! thank you!

**Pull request merged**